test/source/greeter.cpp

```diff
@@ -16,6 +16,11 @@ TEST_CASE("Greeter") {
 }
 
 TEST_CASE("Greeter version") {
+#if (__cpp_lib_starts_ends_with)
+  static_assert(std::string_view(GREETER_VERSION).starts_with("1.0"));  // TBD C++20 only
+  CHECK(std::string(GREETER_VERSION).starts_with("1.0"));               // SameMajorVersion
+#else
   static_assert(std::string_view(GREETER_VERSION) == std::string_view("1.0"));
   CHECK(std::string(GREETER_VERSION) == std::string("1.0"));
-}
\ No newline at end of file
+#endif
+}
```

Comment:
I don't think splitting test cases based on features is a good idea unless the CI tests all possible configurations. The later is definitely out of the scope for this starter.

Reply:
the second real problem: if on other version is found on build host, the test does not compile!